### PR TITLE
Upgrade to Scala 3 and Monocle 3

### PR DESCRIPTION
**Pull request by joan38**
All credits to @inanme

cc @GreyPlane

### CODE REVIEW

**Comment by joan38** [comment]
@GreyPlane 1 more approval needed

File: README.md
Comment (GreyPlane) [inline README.md:19]: probably should remove 'Scala Native'

File: build.sbt
Comment (GreyPlane) [inline build.sbt:8]: i guess Scala 3.3.0 would be better?

**Review by GreyPlane — APPROVED** [review]

**Comment by joan38** [comment]
Addressed the comments.

> 1 workflow awaiting approval

I need approval to run CI:

**Comment by GreyPlane** [comment]
> @GreyPlane Addressed the comments.
> 
> 
> 
> > 1 workflow awaiting approval
> 
> > This workflow requires approval from a maintainer
> 
> 
> 
> I need approval to run CI:

I'm afraid that has to be done by a maintainer and I'm not

**Comment by joan38** [comment]
Who are the maintainers of this repo? @travisbrown ?

**Comment by joan38** [comment]
@zarthross ? Are we good to merge?

**Comment by aheiska** [comment]
Should versioning-table have own columns for 2.12 and 2.13?

**Comment by joan38** [comment]
@aheiska The table is mostly to highlight the Monocle and Circe version it relates to. I let Scala 2 and Scala 3 to do like [circe-extra](https://github.com/circe/circe-generic-extras/blob/main/README.md)

**Comment by joan38** [comment]
Ping @zarthross. Sorry for the insisting push on this, but we are contemplating forking this repo @Netflix to get our teams moving forward.

**Review by zarthross — APPROVED** [review]
Sorry for the delay, this looks good to me.  I think we can get a release cut shortly.

File: README.md
Comment (zarthross) [inline README.md:24]: I don't want to hold up the PR for this change, but we need to split Scala 2 into 2.12, and 2.13 to make more clear that 0.15.0 does not support scala 2.12.

**Comment by zarthross** [comment]
Closes https://github.com/circe/circe-optics/pull/336
Closes https://github.com/circe/circe-optics/pull/231
Closes https://github.com/circe/circe-optics/pull/243

File: README.md
Comment (joan38) [reply README.md:24]: Addressed by https://github.com/circe/circe-optics/commit/c0a6512cf11a9f28030e3fac2bca4d184ea5ee7e

**Comment by joan38** [comment]
👏🏼 Thanks @zarthross @inanme @arixmkii @hughsimpson @GreyPlane for the help

**Comment by zarthross** [comment]
https://github.com/circe/circe-optics/releases/tag/v0.15.0

**Comment by zarthross** [comment]
This should be landing on maven soon, its past CI out in sonatype land now.  Thanks everyone!

**Pull request merged**
